edgelet/edgelet-http/src/lib.rs

```diff
@@ -217,9 +223,37 @@ where
 
         let protocol = Arc::new(protocol);
 
+        let (lock, limit) = if let ConcurrencyThrottling::Limited(limit) = concurrency {
+            (Some(Arc::new(AtomicU32::new(*limit))), *limit)
+        } else {
+            (None, 0 as u32)
+        };
+
         let srv = incoming.for_each(move |(socket, addr)| {
-            let protocol = protocol.clone();
 
+            if let Some(lock) = lock.clone() {
+                let limit_reached = lock.fetch_update( Ordering::AcqRel,
+                    Ordering::Acquire,
+                    |current| {
+                        if current == 0 {
+                            None
+                        } else {
+                            Some(current - 1)
+                        }},
```

Comment:
Just as I said in Gordon's PR: `current.checked_sub(1)`

Reply:
Fixed

Reply:
I'm telling you to replace the whole function body with `current.checked_sub(1)`, not just the `else` block.

Reply:
oh I see, sorry about that.